cmd/otel-allocator/allocation/strategy/strategy.go

```diff
@@ -0,0 +1,122 @@
+package strategy
+
+import (
+	"errors"
+	"fmt"
+	"net/url"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+	"github.com/prometheus/common/model"
+)
+
+type AllocatorProvider func() Allocator
+
+var (
+	registry = map[string]AllocatorProvider{}
+
+	// TargetsPerCollector records how many targets have been assigned to each collector
+	// It is currently the responsibility of the strategy to track this information.
+	TargetsPerCollector = promauto.NewGaugeVec(prometheus.GaugeOpts{
+		Name: "opentelemetry_allocator_targets_per_collector",
+		Help: "The number of targets for each collector.",
+	}, []string{"collector_name"})
+)
+
+func New(name string) (Allocator, error) {
```

Comment:
Sorry if this is obvious or has already been discussed, but what is the benefit of the `init()` registration pattern? Couldn't we pass an interface into `New()` instead and not require global state? Any caller will always have to import their strategy (or strategies) and this package.

Reply:
Having a mapping of string identifiers to strategy implementations enables using the CRD or other configuration mechanisms to select from among multiple available strategies at runtime.  This could be implemented by the consuming system, but would likely result in duplicate implementation.

Reply:
got it, thanks for explaining! 🙂